### PR TITLE
Added podman driver for molecule

### LINKED ISSUE #1572: Buildah/Podman/CRI-O Virtualization Driver
# Issue Type

- Feature request

# Molecule and Ansible details

Ansible 2.7
Molecule 2.19

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- OS package

# Desired Behavior

Similar to how the docker driver is setup for creating the virtualized environment using containers, Buildah (or even possibly Podman) support could provide the same functionality without the need of root privileges or the docker daemon via the use of CRI-O.

```
$ molecule init role --role-name example --driver-name buildah
```

https://docs.ansible.com/ansible/2.7/plugins/connection/buildah.html

### PR DESCRIPTION
**Pull request by sshnaidm**
#### PR Type
- Feature Pull Request
Fixes #1572 

POC of running molecule with podman driver. It may require a patch in ansible which fixes mount problem for rootless podman containers: https://github.com/ansible/ansible/pull/57741 (see issue https://github.com/ansible/ansible/issues/57740)

still requires clean up from docker leftovers and schema (which was copied from dockers one)
Using mostly podman CLI as ansible podman modules don't support yet the required functionality

### CODE REVIEW

**Comment by sshnaidm** [comment]
Ready for review

**Comment by ssbarnea** [comment]
As a note: podmain support in molecule will be considered purely experimental at least until we have it releases in a stable version of ansible.

File: molecule/driver/podman.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
+#  Copyright (c) 2015-2018 Cisco Systems, Inc.
```
[the remark this answers is not in the document]
Comment (sshnaidm) [reply molecule/driver/podman.py]: ha, good catch

**Comment by it-praktyk** [comment]
For reference only: #2158

**Pull request merged**